diff -up gobject-introspection-1.63.2/giscanner/dumper.py gobject-introspection-1.63.2/giscanner/dumper.py-om
--- gobject-introspection-1.63.2/giscanner/dumper.py	2020-01-17 14:55:41.000000000 +0100
+++ gobject-introspection-1.63.2/giscanner/dumper.py-om	2020-03-07 01:53:40.504771928 +0100
@@ -252,6 +252,9 @@ class DumpCompiler(object):
             for ldflag in shlex.split(os.environ.get('LDFLAGS', '')):
                 args.append(ldflag)
 
+         # Make sure we can deal with LTO bytecode in object files...
+         args.append('-flto')
+
         dll_dirs = utils.dll_dirs()
         dll_dirs.add_dll_dirs(self._packages)
 
