From 63c1d1c8d34ea1468acd3cf74f2e5a79d0928049 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jani=20V=C3=A4limaa?= <wally@mageia.org>
Date: Mon, 2 Sep 2019 17:07:09 +0300
Subject: [PATCH] Fix linking against python3

---
 giscanner/meson.build | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/giscanner/meson.build b/giscanner/meson.build
index a3b06f3..8a4dbc3 100644
--- a/giscanner/meson.build
+++ b/giscanner/meson.build
@@ -100,10 +100,10 @@ giscanner_lib = static_library('giscanner',
 )
 
 # https://github.com/mesonbuild/meson/issues/4117
-if host_machine.system() == 'windows'
-  python_ext_dep = python.dependency()
-else
-  python_ext_dep = python.dependency().partial_dependency(compile_args: true)
+python_ext_dep = dependency('python3-embed', required: false)
+
+if not python_ext_dep.found()
+  python_ext_dep = dependency('python3', required: false)
 endif
 
 if not python_ext_dep.found()
-- 
2.22.1
